pkg/cosi/cosi_bucket_class.go

```diff
@@ -0,0 +1,435 @@
+package cosi
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	nbv1 "github.com/noobaa/noobaa-operator/v5/pkg/apis/noobaa/v1alpha1"
+	"github.com/noobaa/noobaa-operator/v5/pkg/bucketclass"
+	"github.com/noobaa/noobaa-operator/v5/pkg/bundle"
+	"github.com/noobaa/noobaa-operator/v5/pkg/options"
+	"github.com/noobaa/noobaa-operator/v5/pkg/util"
+	"github.com/noobaa/noobaa-operator/v5/pkg/validations"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CmdCOSIBucketClass returns a CLI command
+func CmdCOSIBucketClass() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "bucketclass",
+		Short: "Manage cosi bucket classes",
+	}
+	cmd.AddCommand(
+		CmdCreateBucketClass(),
+		CmdDeleteBucketClass(),
+		CmdStatusBucketClass(),
+		CmdListBucketClass(),
+	)
+	return cmd
+}
+
+// CmdCreateBucketClass returns a CLI command
+func CmdCreateBucketClass() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create an cosi bucket class",
+	}
+
+	cmd.AddCommand(
+		CmdCreateNamespaceBucketclass(),
+		CmdCreatePlacementBucketClass(),
+	)
+
+	return cmd
+}
+
+// CmdCreatePlacementBucketClass returns a CLI command
+func CmdCreatePlacementBucketClass() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "placement-bucketclass <bucket-class-name>",
+		Short: "Create placement policy bucket class",
+		Run:   RunCreatePlacementBucketClass,
+	}
+
+	// placement policy flags
+	cmd.Flags().String("placement", "",
+		"Set first tier placement policy - Mirror | Spread | \"\" (empty defaults to single backing store)")
+	cmd.Flags().StringSlice("backingstores", nil,
+		"Set first tier backing stores (use commas or multiple flags)")
+	cmd.Flags().String("replication-policy", "",
+		"Set the json file name that contains the replication rules")
+	cmd.Flags().String("deletion-policy", "delete",
+		"Set the deletion policy of the buckets delete | retain - the default deletion policy is delete")
```

Comment:
maybe rephrase this so user will better understand the meaning of the options and the default.

Reply:
I also changed the default to retain as COSI's default